gufe/protocols/protocolunit.py

```diff
@@ -302,22 +304,27 @@ def execute(self, *,
             result = ProtocolUnitResult(
                 name=self.name, source_key=self.key, inputs=inputs, outputs=outputs
             )
-
         except Exception as e:
             if raise_error:
                 raise
 
+            outputs = {}
+
             result = ProtocolUnitFailure(
                 name=self._name,
                 source_key=self.key,
                 inputs=inputs,
-                outputs=dict(),
+                outputs=outputs,
                 exception=(e.__class__.__qualname__, e.args),
                 traceback=traceback.format_exc()
             )
 
-        # TODO: change this part once we have clearer ideas on how to inject
-        # persistent storage use
+        output_files = [f for f in outputs.values() if isinstance(f, Path)]
+        for path in output_files:
+            # label is worth further consideration before merge
+            label = str(self.key / path)
+            shared.store_path(label, path)
+
```

Comment:
so this is scanning the outputs for files that are indicated in the outputs, then explicitly storing them in the `ExternalResource`? if so this deserves a comment to explain that

then also a test to check that this is working